Terrain Xml Cleanup

SgtBears opened the pull request:
Can only be merged after the text changes.

CIV4BonusInfos.xml:     Removed unneccessary <iYieldChange>0
                Removed TechCityTrade<TECH_MINING for Coal, Copper and Iron since Steam Power, Bronze Working and Iron Working all require Mining.
                Removed TechCityTrade<TECH_ANIMAL_HUSBANDRY for Horse for reasons above
                Moved extra banana food to Improvement

CIV4FeatureInfos.xml:       Removed unneccessary <iYieldChange>0
                Removed Rhye Notes
                Corrected Text Keys
                Made Mud, Cape and Islands Nuke Immune
                Made Islands and Cape disable city founding (like all sea terrains)

CIV4ImprovementInfos.xml:   Removed Optional Tags
                Removed Unneccessary <iYield>0
                Moved High Sea Fishing Boats Extra food and Slave Plantation Extra Production and Commerce to the Improvement instead of the bonus (Makes it easier to tell what the difference between the base improvement and the new one is)
                Corrected Text Keys
                Moved extra banana food to Improvement
                Removed Rhye Notes

CIV4TerrainInfos.xml:       Removed Unneccessary <iYield>0
                Removed Rhye Notes
                Corrected Text Keys
                Added cannot found cities to Marsh
                Removed Pointless Notes
                Removed Optional Tags

SgtBears committed:
CIV4BonusInfos.xml:		Removed unneccessary <iYieldChange>0
				Removed TechCityTrade<TECH_MINING for Coal, Copper and Iron since Steam Power, Bronze Working and Iron Working all require Mining.
				Removed TechCityTrade<TECH_ANIMAL_HUSBANDRY for Horse for reasons above
				Moved extra banana food to Improvement

CIV4FeatureInfos.xml:		Removed unneccessary <iYieldChange>0
				Removed Rhye Notes
				Corrected Text Keys
				Made Mud, Cape and Islands Nuke Immune
				Made Islands and Cape disable city founding (like all sea terrains)

CIV4ImprovementInfos.xml:	Removed Optional Tags
				Removed Unneccessary <iYield>0
				Moved High Sea Fishing Boats Extra food and Slave Plantation Extra Production and Commerce to the Improvement instead of the bonus (Makes it easier to tell what the difference between the base improvement and the new one is)
				Corrected Text Keys
				Moved extra banana food to Improvement
				Removed Rhye Notes

CIV4TerrainInfos.xml:		Removed Unneccessary <iYield>0
				Removed Rhye Notes
				Corrected Text Keys
				Added cannot found cities to Marsh
				Removed Pointless Notes
				Removed Optional Tags

dguenms commented on Assets/XML/Terrain/CIV4BonusInfos.xml:
```diff
@@ -72,13 +71,12 @@
 			<BonusClassType>BONUSCLASS_GENERAL</BonusClassType>
 			<ArtDefineTag>ART_DEF_BONUS_COAL</ArtDefineTag>
 			<TechReveal>TECH_STEAM_POWER</TechReveal>
-			<TechCityTrade>TECH_MINING</TechCityTrade>
```
Please keep.

dguenms commented on Assets/XML/Terrain/CIV4BonusInfos.xml:
```diff
@@ -126,13 +124,12 @@
 			<BonusClassType>BONUSCLASS_RUSH</BonusClassType>
 			<ArtDefineTag>ART_DEF_BONUS_COPPER</ArtDefineTag>
 			<TechReveal>TECH_BRONZE_WORKING</TechReveal>
-			<TechCityTrade>TECH_MINING</TechCityTrade>
```
sa

dguenms commented on Assets/XML/Terrain/CIV4BonusInfos.xml:
```diff
@@ -192,13 +189,12 @@
 			<BonusClassType>BONUSCLASS_RUSH</BonusClassType>
 			<ArtDefineTag>ART_DEF_BONUS_HORSE</ArtDefineTag>
 			<TechReveal>TECH_ANIMAL_HUSBANDRY</TechReveal>
-			<TechCityTrade>TECH_ANIMAL_HUSBANDRY</TechCityTrade>
```
sa

dguenms commented on Assets/XML/Terrain/CIV4BonusInfos.xml:
```diff
@@ -250,13 +246,12 @@
 			<BonusClassType>BONUSCLASS_RUSH</BonusClassType>
 			<ArtDefineTag>ART_DEF_BONUS_IRON</ArtDefineTag>
 			<TechReveal>TECH_IRON_WORKING</TechReveal>
-			<TechCityTrade>TECH_MINING</TechCityTrade>
```
sa

dguenms commented on Assets/XML/Terrain/CIV4ImprovementInfos.xml:
```diff
@@ -150,34 +147,8 @@
 			<iHappiness>0</iHappiness>
 			<iPillageGold>0</iPillageGold>
 			<bOutsideBorders>0</bOutsideBorders>
-			<TerrainMakesValids>
```
Keep this section.

dguenms commented on Assets/XML/Terrain/CIV4ImprovementInfos.xml:
```diff
@@ -374,6 +327,9 @@
 			<Description>TXT_KEY_IMPROVEMENT_HIGH_SEA_FISHING_BOATS</Description>
 			<Civilopedia>TXT_KEY_IMPROVEMENT_FISHING_BOATS_PEDIA</Civilopedia>
 			<ArtDefineTag>ART_DEF_IMPROVEMENT_FISHING_BOATS</ArtDefineTag>
+			<YieldChanges>
```
I don't see a reason to split it in this way, please revert.

dguenms commented on Assets/XML/Terrain/CIV4ImprovementInfos.xml:
```diff
@@ -934,9 +869,7 @@
-						<iYieldChange>2</iYieldChange>
-						<iYieldChange>0</iYieldChange>
-						<iYieldChange>0</iYieldChange>
+						<iYieldChange>3</iYieldChange>
```
Why from 2 to 3?

dguenms commented on Assets/XML/Terrain/CIV4ImprovementInfos.xml:
```diff
@@ -1061,6 +992,11 @@
 			<Description>TXT_KEY_IMPROVEMENT_SLAVE_PLANTATION</Description>
 			<Civilopedia>TXT_KEY_IMPROVEMENT_SLAVE_PLANTATION_PEDIA</Civilopedia>
 			<ArtDefineTag>ART_DEF_IMPROVEMENT_PLANTATION</ArtDefineTag>
+			<YieldChanges>
```
sa

dguenms commented on Assets/XML/Terrain/CIV4ImprovementInfos.xml:
```diff
@@ -1874,9 +1794,8 @@
 			<Type>IMPROVEMENT_FOREST_PRESERVE</Type>
 			<Description>TXT_KEY_IMPROVEMENT_FOREST_PRESERVE</Description>
-			<Civilopedia>TXT_KEY_IMPROVEMENT_FOREST_PRESERVE_PEDIA</Civilopedia>
+			<Civilopedia>TXT_KEY_IMPROVEMENT_FOREST_PRESERVE_IMPROVEMENT_PEDIA</Civilopedia>
```
Improvement twice in the key?

SgtBears committed:
Merge branch 'develop' of https://github.com/dguenms/Dawn-of-Civilization into Terrain_Xml

SgtBears commented:
Fixed your concerns. Also added Rainforest to the FeatureMakesValids of the goody hut.

dguenms commented: Why from 2 to 3?
I also decreased bananas base yield from 2 to 1. It's so it matches every other bonus and we went through this with the whale bonus. I can revert it if you want.

dguenms commented: Improvement twice in the key?
There was already a TXT_KEY_IMPROVEMENT_FOREST_PRESERVE_PEDIA which refered to something else so I changed the forest preserves pedia text key definition.

SgtBears committed:
Fixes

dguenms commented:
I see on both counts. I agree that the banana bonus/improvement change is more sensible now.
